app/pathfinder_svc.py

```diff
@@ -73,6 +74,110 @@ def add_fact(fact_list, trait, value):
+        return r
+
+    async def generate_exploitability_graph(self, report):
+        """"""
+        def _is_edge_exploitable(owned, candidate):
+            """
+                Determines if candidate node is exploitable from current node. Exploitability on the
+                target node is defined as obtaining presence (either normal or privileged access) on the
+                candidate node.
+            Args:
+                owned (): source node to determine exploitability from
+                candidate (): target node to determine exploitability to
+            Returns: 
+                (bool)
+            """
+            # Steps:
+            # - Is candidate node/ability a freebie?
+            # - return True
+            # - Is candidate node/ability probabalistic?
+            # - return True
+            # - Is candidate node/ability blacklisted?
+            # - return False
+            # - Does current node have an ability to laterally move to the candidate node? Or does it have a probabilitic 
+            # - If lateral movement ability present, are pre conditions met to use it?
+            # - return True
+            if candidate.is_denied():
+                return False
+            if not candidate.can_access():
+                return False
+            if candidate.freebie_abilities:
+                return True
+            if candidate.cves:
+                return True
+            return False
+
+        exploit_map = nx.Graph()
+        for edge_ in report.network_map.edges:
+            source_node = report.retrieve_host_by_id(edge_[0])
+            target_node = report.retrieve_host_by_id(edge_[1])
+            if _is_edge_exploitable(source_node, target_node):
+                exploit_map.add_node(source_node)
+                exploit_map.add_node(target_node)
+                exploit_map.add_edge(source_node, target_node)
+        return exploit_map
+
+    async def generate_exploitable_paths(self, report, exploitability_graph, source, target):
+        """Find all paths from source to target for the given exploitability graph.
+        Args:
+            exploitability_graph (networkx.Graph)
+            source (node):
+            target (node):
+        Returns: 
+            (generator) list of paths
+        """
+        ret = list()
+        if (not report.retrieve_host_by_id(source) in exploitability_graph.nodes or 
+            not report.retrieve_host_by_id(target) in exploitability_graph.nodes):
+            return None
+        paths = nx.all_simple_paths(exploitability_graph, report.retrieve_host_by_id(source), report.retrieve_host_by_id(target))
+        # create adversaries for every path
+        for path in paths:
+            ret.append(dict(path=path, adversary=self.create_adversary_from_path(report, path)))
+        return ret
+
+    def create_adversary_from_path(self, report, path):
```

Comment:
I dont see changes to `gather_techniques()` to actually carry out the design as stated here.

 i.e. how would `(NodeFreebie, "")` instances be added to the adversary path?